interop/test/ping.spec.ts

```diff
@@ -0,0 +1,164 @@
+/* eslint-disable no-console */
+/* eslint-env mocha */
+
+import { } from 'aegir/chai'
+import { createLibp2p, Libp2pOptions } from 'libp2p'
+import { webTransport } from '@libp2p/webtransport'
+import { tcp } from '@libp2p/tcp'
+import { webSockets } from '@libp2p/websockets'
+import { noise } from '@chainsafe/libp2p-noise'
+import { mplex } from '@libp2p/mplex'
+import { yamux } from '@chainsafe/libp2p-yamux'
+import { multiaddr } from '@multiformats/multiaddr'
+import { webRTC } from '@libp2p/webrtc'
+import { pingService, PingService } from 'libp2p/ping'
+
+async function redisProxy (commands: any[]): Promise<any> {
+  const res = await fetch(`http://localhost:${process.env.proxyPort ?? ''}/`, { body: JSON.stringify(commands), method: 'POST' })
+  if (!res.ok) {
+    throw new Error('Redis command failed')
+  }
+  return await res.json()
+}
+
+describe('ping test', () => {
+  // eslint-disable-next-line complexity
+  it('should ping', async () => {
+    const TRANSPORT = process.env.transport
+    const SECURE_CHANNEL = process.env.security
+    const MUXER = process.env.muxer
+    const isDialer = process.env.is_dialer === 'true'
+    const IP = process.env.ip ?? '0.0.0.0'
+    const timeoutSecs: string = process.env.test_timeout_secs ?? '180'
+
+    const options: Libp2pOptions<{ ping: PingService }> = {
```

Comment:
Can all this config be built in a separate file and invoked in a `beforeEach` block?

It's very verbose and doesn't seem like it's part of the test.

Reply:
I don't see the benefit of having a separate file, but I can move to a beforeEach.

Reply:
Presumably these tests will test more than just ping in the future, at that point we're going to want to set up a libp2p node in more than one place, hence the external file.

We can make that refactor when we need it, but also this is setup code and not test code so it shouldn't be in the `it` function.

Reply:
This is testing whether two libp2p nodes can communicate. A future test may look similar, but I don’t think it makes sense to refactor until we have that future test.